src/plugin.js

```diff
@@ -85,6 +86,19 @@ const initPlugin = function(player, options) {
 
   // clears the previous monitor timeout and sets up a new one
   const resetMonitor = function() {
+    // start the loading spinner if player has stalled
+    window.clearTimeout(waiting);
+    player.removeClass('vjs-waiting');
```

Comment:
I'm not sure I like the idea of potentially changing the DOM on `timeupdate`. That said, it may be infrequent enough to not matter. Maybe we could track the state in a variable and only add/remove the class when necessary?